### DISCUSSION TITLE
How do I get the current locale language used by nls on the backend, and nls.locale cannot be used on the backend?

### DISCUSSION THREAD

**Discussion by noneghost** [eclipse-theia/theia#13470] [Q&A] [answered]
The window API used in nls.locale cannot be used on the backend. Is there any other way to get the translation locale used by the frontend on the backend?

https://github.com/eclipse-theia/theia/blob/7012fd29fa9d3e78327e4f0d678ef4bae58c15e8/packages/core/src/common/nls.ts#L27

**Comment by msujew** [comment, marked as answer]
Hey @noneghost,

The `LocalizationProvider` backend service contains the locale of the most recently connected frontend. This should be accurate enough to localize messages on the backend.

**Reply by noneghost** [reply]
Thankfully, this is the effective method.
